public/docs/_examples/cb-validation/ts/app/hero-form-model-binding.component.html

```diff
@@ -0,0 +1,75 @@
+ <!-- #docplaster -->
+ <!-- #docregion -->
+<div class="container">
+  <div [hidden]="submitted">
+    <h1>Hero Form (Model-Driven with Data Binding)</h1>
+    <form [ngFormModel]="heroForm" *ngIf="active" 
+          (ngSubmit)="onSubmit()">
+      <div class="form-group">
+        <!-- #docregion name-with-error-msg -->
+        <label for="name">Name</label>
+        <input type="text" class="form-control" 
+               [(ngModel)]="model.name"
+               ngControl="name"
+               [ngClass]="{'required': isRequired('name')}">
```

Comment:
Because the validation is defined in the code, it could be changed based on the user or situation. For example, a field may be required for a normal user, but not an admin user. When these validation rules are in the code, there is no easy way to put the validation aria attributes in the HTML and have them work properly. (I supposed property binding could work ... but is messy in this case.)
Just as the dev puts the basic attributes in the HTML but the validation rules in the code, it would make sense for the dev to put the basic aria attributes in the HTML, but the validation aria info would be defined in the code. Angular should then add the appropriate validation *and* aria attributes to the HTML automatically.